### ISSUE TITLE
Configure Health Check Interval

### ISSUE THREAD

**Issue by Zaxim** [haugene/docker-transmission-openvpn#771]
**Is your feature request related to a problem? Please describe.**
When I start a haugene/transmission-openvpn it remains in a `(health: starting)` STATUS for 5 minutes, until the first health check triggers. Because the STATUS is not `(healthy)` my traefik reverse proxy will not create routes to the backend and the container is not accessible for 5 minutes.

**Describe the solution you'd like**

I would like to be able to configure how often the health check is performed, as opposed to the currently hard-coded 5 minutes. I could potentially shorten it to 2 minutes with no ill-effect and my container would be accessible sooner.

This could also be used be people who want to decrease the frequency of the health check, without disabling it completely.

**Describe alternatives you've considered**

A specific fix to my problem would be to force the health check to attempt immediately after the transmission scripts complete, as opposed to waiting to the scheduled time (I presume you're using cron).

**Comment by gdlx** [comment]
@Zaxim Did you find a workaround ? I thought some Traefik label would allow to ignor the container is starting but I don't find any...

**Comment by Zaxim** [comment]
Unfortunately, I have not found a workaround but I haven't dug too deeply into it since I opened the issue.

**Comment by gdlx** [comment]
I've found one actually, not on the traefik side, but just in the docker run command, by using `--health-interval` and `--health-start-period` to override `Dockerfile` config.

I've set `--health-interval` to 5s and `--health-start-period` to 10s, which makes the container available after 15 seconds. It works fine in my case but it may require tuning depending on configuration and connection speed.

5 minutes by default seems way too much IMO anyway (is there any healthy case where the VPN takes 5 minutes to be up ?). Maybe there's a good reason I don't know, but it's finally really easy to override anyway.

**Comment by haugene** [comment]
Closing this with the solution proposed by @gdlx and adding a link to the documentation for this in the Docker Engine: https://docs.docker.com/engine/reference/run/#healthcheck

I do agree though, that 5 minutes is too long. We didn't think of the Traefik-usecase when we set it up. The reason for adding the healthcheck was to detect when the container lost connection to the internet (which apparently happens for some providers/hosts) and then being able to flag it so the the container could be restarted.

Maybe we'll change it going forward - but for now the Traefikers just need to override it.
